packages/connection-form/src/components/connection-form.tsx

```diff
@@ -97,6 +108,165 @@ type ConnectionFormPropsWithoutPreferences = {
   onSaveConnectionClicked: (connectionInfo: ConnectionInfo) => Promise<void>;
 };
 
+const colorPreviewStyles = css({
+  height: '16px',
+  width: '16px',
+  marginRight: spacing[2],
+});
+
+const ColorCircleGlyph = createGlyphComponent('ColorCircle', (props) => (
+  <svg
+    {...props}
+    className={colorPreviewStyles}
+    width="16"
+    height="16"
+    viewBox="0 0 16 16"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+  >
+    <g clip-path="url(#clip0_756_18092)">
+      <line
+        opacity="0.5"
+        x1="16.3536"
+        y1="-0.146447"
+        x2="0.353554"
+        y2="15.8536"
+        stroke="#DB3030"
+      />
+    </g>
+    <rect
+      x="0.5"
+      y="0.5"
+      width="15"
+      height="15"
+      rx="7.5"
+      stroke="#889397"
+      fill={props.hexColor}
+    />
+    <defs>
+      <clipPath id="clip0_756_18092">
+        <rect width="16" height="16" rx="8" fill="white" />
+      </clipPath>
+    </defs>
+  </svg>
+));
+
+const personalisationSectionLayoutStyles = css({
+  display: 'grid',
+  gridTemplateColumns: '1fr 1fr',
+  gridTemplateRows: 'auto',
+  gridTemplateAreas: `
+    'name-input color-input'
+    'favorite-marker favorite-marker'
+  `,
+  gap: spacing[4],
+  marginBottom: spacing[4],
+});
+
+type ConnectionPersonalisationFormProps = {
+  initialValue?: ConnectionPersonalisationOptions;
+  onChange: (ConnectionPersonalisationOptions) => void;
+};
+
+function ConnectionPersonalisationForm({
+  onChange,
+  initialValue,
+}: ConnectionPersonalisationFormProps): React.ReactElement {
+  const showFavoriteActions = useConnectionFormPreference(
+    'showFavoriteActions'
+  );
+
+  const [name, setName] = useState(initialValue?.name || '');
+  const [isNameDirty, setNameDirty] = useState(!!initialValue?.isNameDirty);
+  const [color, setColor] = useState(initialValue?.color || undefined);
+  const [isFavorite, setFavorite] = useState(initialValue?.isFavorite);
+
+  const onChangeName = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      const name = ev.target.value;
+
+      setName(name);
+      setNameDirty(true);
+
+      onChange({ name, color, isFavorite, isNameDirty: true });
+    },
+    [color, isFavorite, setName, setNameDirty]
+  );
+
+  const onChangeColor = useCallback(
+    (newValue: string) => {
+      const color = newValue === 'no-color' ? undefined : newValue;
+
+      setColor(color);
+      onChange({ name, color, isFavorite, isNameDirty });
+    },
+    [name, isFavorite, isNameDirty, setColor]
+  );
+
+  const onChangeFavorite = useCallback(
+    (ev: React.ChangeEvent<HTMLInputElement>) => {
+      const isFavorite = ev.target.checked;
+
+      setFavorite(isFavorite);
+      onChange({ name, color, isFavorite, isNameDirty });
+    },
+    [name, color, isNameDirty, setFavorite]
+  );
+
+  useEffect(() => {
+    setName(initialValue?.name);
+  }, [initialValue, setName]);
```

Comment:
If this is an initial value, it probably shouldn't unpredictably change component state when prop updates?

Reply:
Yeah you are right, now it's not necessary anymore. Now this state is handled in the react reducer, and the naming is better. Is not only "initial state" but it's synchronised when the connection url changes and the name field is not dirty. Now the code should be more approachable and easy to understand.